src/test/java/com/philips/research/bombar/core/domain/licenses/ObligationAnalyzerTest.java

```diff
@@ -0,0 +1,124 @@
+package com.philips.research.bombar.core.domain.licenses;
+
+import com.philips.research.bombar.core.domain.Dependency;
+import com.philips.research.bombar.core.domain.Project;
+import com.philips.research.bombar.core.domain.Relation;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ObligationAnalyzerTest {
+    private static final String OTHER_OBLIGATION = "Other";
+    private static final String OTHER_OBLIGATION_DESC = "Other desc";
+    private static final String KEY = "Key";
+    private static final String TITLE = "Title";
+    private static final String LICENSE = "License";
+    private static final String OBLIGATION = "Obligation";
+    private static final String OBLIGATION_DESC = "Obligation desc";
+
+    private final LicenseRegistry registry = new LicenseRegistry();
+    private final Project project = new Project(UUID.randomUUID());
+    private final ObligationAnalyzer analyzer = new ObligationAnalyzer(registry, project);
+
+    @BeforeEach
+    void setUp() {
+        registry.term(OBLIGATION, OBLIGATION_DESC);
+        registry.term(OTHER_OBLIGATION, OTHER_OBLIGATION_DESC);
+    }
+
+    @Test
+    void noObligationForEmptyProject() {
+        final var obligations = analyzer.findObligations();
+
+        assertThat(obligations).isEmpty();
+    }
+
+    @Test
+    void noObligationForUnknownLicense() {
+        final var dependency = new Dependency(KEY, TITLE).setLicense("unknown");
+        project.addDependency(dependency);
+
+        final var obligations = analyzer.findObligations();
+
+        assertThat(obligations).isEqualTo(Map.of("Unknown Obligations", Set.of(dependency)));
+    }
+
+    @Test
+    void obligationForLicense() {
+        registry.license(LICENSE).requires(OBLIGATION);
+        final var dependency = new Dependency(KEY, TITLE).setLicense(LICENSE);
```

Comment:
Creating a dependency and adding it to the project is repeated in most tests now.

Consider grouping these tests, and settings this up only once. (An alternative could be to at least declare and define a `private final Dependency dependency = new Dependency(KEY, TITLE)` at the top of the class.)